status/queues.py

```diff
@@ -8,72 +8,79 @@
 from genologics import lims
 from genologics.entities import Queue, Artifact, Container
 import xml.etree.ElementTree as ET
+import psycopg2
 
 
 class qPCRPoolsDataHandler(SafeHandler):
     """ Serves a page with qPCR queues from LIMS listed
     URL: /api/v1/qpcr_pools
     """
     def get(self):
-        limsl = lims.Lims(BASEURI, USERNAME, PASSWORD)
-        #qPCR queues
-        queues = {}
-        queues['MiSeq'] = Queue(limsl, id='1002')
-        queues['NovaSeq'] = Queue(limsl, id='1666')
-        queues['LibraryValidation'] = Queue(limsl, id='41')
-
-        methods = queues.keys()
-        pools = {}
         qpcr_control_names = [ 'AM7852', 'E.Coli genDNA', 'Endogenous Positive Control', 'Exogenous Positive Control',
                                 'Human Brain Reference RNA', 'lambda DNA', 'mQ Negative Control', 'NA10860', 'NA11992',
                                 'NA11993', 'NA12878', 'NA12891', 'NA12892', 'No Amplification Control',
                                 'No Reverse Transcriptase Control', 'No Template Control', 'PhiX v3', 'Universal Human Reference RNA',
                                 'lambda DNA (qPCR)'
                               ]
-        for method in methods:
-            pools[method] ={}
-            if queues[method].artifacts:
-                tree = ET.fromstring(queues[method].xml())
-                if tree.find('next-page') is not None:
-                    flag =True
-                    next_page_uri = tree.find('next-page').attrib['uri']
-                    while flag:
-                        next_page = ET.fromstring(Queue(limsl, uri = next_page_uri).xml())
-                        for elem in next_page.findall('artifacts'):
-                            tree.insert(0, elem)
-                        if next_page.find('next-page') is not None:
-                            next_page_uri = next_page.find('next-page').attrib['uri']
-                        else:
-                            flag = False
-                for artifact in tree.iter('artifact'):
-                    queue_time = artifact.find('queue-time').text
-                    container = Container(limsl, uri = artifact.find('location').find('container').attrib['uri']).name
-                    art = Artifact(limsl, uri = artifact.attrib['uri'])
-                    value = artifact.find('location').find('value').text
-                    library_type = ''
-                    runmode = ''
-
-                    #skip if the Artifact is a control
-                    if art.name in qpcr_control_names:
-                        continue
+        queues = {}
```

Comment:
Hi @aanil, do you think we could also add the qPCR step in the NextSeq protocol in the queue?

Reply:
Done!

Reply:
👍